code/game/machinery/computer/robot.dm

```diff
@@ -41,228 +41,133 @@
-	onclose(user, "computer")
-	return
+/obj/machinery/computer/robotics/ui_data(mob/user)
+	var/list/data = list()
 
-/obj/machinery/computer/robotics/Topic(href, href_list)
-	if(..())
-		return 1
-	else
-		usr.set_machine(src)
+	data["can_hack"] = FALSE
+	if((isAI(user) && ismalf(user)) || isAdminGhost(user))
+		data["can_hack"] = TRUE
+	if(hacking)
+		data["can_hack"] = FALSE
 
-		if (href_list["eject"])
-			temp = {"
-			Start Robot Destruction Sequence?<BR>
-			<BR><A href='?src=\ref[src];eject2=1'>Yes</A><BR>
-			<A href='?src=\ref[src];temp=1'>No</A>"}
+	data["sequence_activated"] = FALSE
+	if(cyborg_detonation_time > world.time)
+		data["sequence_activated"] = TRUE
+	data["sequence_timeleft"] = (cyborg_detonation_time-world.time)/10
 
-		else if (href_list["eject2"])
-			if (cyborg_detonation_time == 0 || cyborg_detonation_time < world.time)
-				message_admins("<span class='notice'>[key_name_admin(usr)] has initiated the global cyborg killswitch!</span>")
-				log_game("<span class='notice'>[key_name(usr)] has initiated the global cyborg killswitch!</span>")
-				start_sequence()
-			temp = null
-
-
-
-		else if (href_list["stop"])
-			temp = {"
-			Stop Robot Destruction Sequence?<BR>
-			<BR><A href='?src=\ref[src];stop2=1'>Yes</A><BR>
-			<A href='?src=\ref[src];temp=1'>No</A>"}
-
-		else if (href_list["stop2"])
-			message_admins("<span class='notice'>[key_name_admin(usr)] has halted the global cyborg killswitch!</span>")
-			log_game("<span class='notice'>[key_name(usr)] has halted the global cyborg killswitch!</span>")
-			stop_sequence()
-			temp = null
-
-		else if (href_list["temp"])
-			temp = null
-		else if (href_list["screen"])
-			switch(href_list["screen"])
-				if("0")
-					screen = 0
-				if("1")
-					screen = 1
-				if("2")
-					screen = 2
-		else if (href_list["killbot"])
-			if(allowed(usr))
-				var/mob/living/silicon/robot/R = locate(href_list["killbot"])
-				if(R)
-					if(istype(usr, /mob/living/silicon/ai))
-						if (R.connected_ai != usr)
-							return
-					if(istype(usr, /mob/living/silicon/robot))
-						if (R != usr)
-							return
-					if(R.scrambledcodes)
-						return
-					var/choice = input("Are you certain you wish to detonate [R.name]?") in list("Confirm", "Abort")
-					if(choice == "Confirm")
-						if(R && istype(R))
-							if(R.self_destruct())
-								message_admins("<span class='notice'>[key_name_admin(usr)] detonated [R.name]!</span>")
-								log_game("<span class='notice'>[key_name_admin(usr)] detonated [R.name]!</span>")
-			else
-				to_chat(usr, "<span class='warning'>Access Denied.</span>")
-		else if (href_list["lockbot"])
+	data["cyborgs"] = list()
+	for(var/mob/living/silicon/robot/R in cyborg_list)
+		if(!can_control(R,user))
+			continue
+		var/list/cyborg_data = list(
+			name = R.name,
+			locked_down = R.lockdown,
+			status = R.stat,
+			charge = R.cell ? R.cell.percent() : null,
+			module = R.module ? "[R.module.name] Module" : "No Module Installed",
```

Comment:
https://github.com/vgstation-coders/vgstation13/blob/ab73db104137216e99d51bc9e5398ed7957c5f33/__DEFINES/silicon.dm#L1-L48

Reply:
```suggestion
			module = R.module ? "[R.modtype] Module" : "No Module Installed",
```
Disregard defines and lists, acquire speed